src/main/java/org/sonarlint/intellij/analysis/AnalysisSubmitter.java

```diff
@@ -139,17 +147,20 @@ private static boolean shouldExecuteInBackground(AnActionEvent e) {
   }
 
   private Cancelable analyzeInBackground(Collection<VirtualFile> files, TriggerType trigger, AnalysisCallback callback) {
-    var task = new Analysis(project, files, trigger, false, callback);
-    TaskRunnerKt.startBackgroundTask(project, ANALYSIS_TASK_TITLE, task::run);
-    return task;
+    if (shouldSkipAnalysis()) {
```

Comment:
I am not sure about this check here. It means if there is a long analysis running and users change some code, it won't be analyzed